### PR TITLE
Improve `CONTRIBUTING.md` and add `CODEOWNERS` file

### PR DESCRIPTION
**Pull request by mrjones2014**
Resolves: #143 

## How to Test

1. Review the contents of `CONTRIBUTING.md` and let me know if there's anything I missed or that could be clarified further

## Potential Side Effects/Things to Look Out For

I will be a required reviewer for all PRs since I added myself as the global code owner in `CODEOWNERS`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,20 +1,69 @@
...
+
+## Submitting Pull Requests
+
+When submitting a Pull Request, **make sure you fill out the Pull Request template.** Pull Requests without this filled out will not be reviewed until it is filled out. If you exclude any part of it,
+you should provide a valid explanation as to why that part is not applicable. Please also make sure test your own code with the provided testing instructions before requesting a review.
+
+Please do not submit Pull Requests which do not resolve an open issue. If you would like to make a Pull Request for which no open issue corresponds, please open the issue. Every Pull Request **must resolve an open issue.**
+
+### Generating a Changelog Entry
+
```
Comment (hsheikhali1) [inline CONTRIBUTING.md:53]: @mrjones2014 Is there a place I can check all changelog types we use for this project? Will help not having to guess

Comment (mrjones2014) [reply CONTRIBUTING.md:53]: Ah, yeah we're using the defaults so they could be found in `COMMANDS.md`. I can update this.

Comment (hsheikhali1) [reply CONTRIBUTING.md:53]: I don't mind going to `COMMANDS.md` but new folks would have no idea and would assume they are in the `CONTRIBUTING.md` file

**Comment by mrjones2014** [comment]
@hsheikhali1 updated.

**Review by hsheikhali1 — APPROVED** [review]
awesome! thanks for adding the change types!

**Pull request merged**